src/main/java/sirius/kernel/async/ParallelTaskExecutor.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.kernel.async;
+
+import sirius.kernel.commons.Wait;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Executes tasks in parallel in the current node using virtual threads with a limit on concurrency.
+ * <p>
+ * After the executor has been initialized, submit tasks using {@link #submitTask(Runnable)}. They will start as
+ * soon as the first task is submitted.
+ * Once all tasks have been submitted, call {@link #shutdownWhenDone()} to wait for all pending tasks to complete.
+ * <p>
+ * Note that the current {@linkplain CallContext#getCurrent() current context} will be passed to the tasks when they
+ * are executed.
+ */
+public class ParallelTaskExecutor {
+
+    private final ExecutorService executor;
+    private final BlockingQueue<Runnable> taskQueue;
+    private final Semaphore semaphore;
+    private final AtomicInteger taskCount;
+    private final CallContext currentContext;
+
+    /**
+     * Creates a new parallel task executor.
+     *
+     * @param maxConcurrentTasks the maximum number of tasks to run concurrently
+     */
+    public ParallelTaskExecutor(int maxConcurrentTasks) {
+        this.currentContext = CallContext.getCurrent();
+        this.executor = Executors.newVirtualThreadPerTaskExecutor();
+        this.taskQueue = new LinkedBlockingQueue<>();
```

Comment:
Is it okay that the queue can still grow indefinitely since `LinkedBlockingQueue` is unbounded.
```
this.taskQueue = new LinkedBlockingQueue<>(maxConcurrentTasks * 2); // Limit queue size?
```

Reply:
For the expected usage yes. If the need arises, we can add a different constructor to cope with a maximum queue size. For now I can add a hint to the JavaDoc explaining that the queue is only limited by your own RAM.